pkg/api/v2/endpoints.go

```diff
@@ -277,3 +283,113 @@ func (s *Service) ListWebhooks(ctx context.Context, req *apiv2.ListWebhooksReque
 func (s *Service) PatchEnv(ctx context.Context, req *apiv2.PatchEnvRequest) (*apiv2.PatchEnvsResponse, error) {
 	return nil, s.base.NewError(http.StatusNotImplemented, apiv2base.ErrorNotImplemented, "Environments not implemented in OSS")
 }
+
+func (s *Service) BulkRerun(ctx context.Context, req *apiv2.BulkRerunRequest) (*apiv2.BulkRerunResponse, error) {
+	// Validate request
+	if len(req.RunIds) == 0 {
+		return nil, s.base.NewError(http.StatusBadRequest, apiv2base.ErrorMissingField, "At least one run ID is required")
+	}
+
+	if len(req.RunIds) > 50 {
+		return nil, s.base.NewError(http.StatusBadRequest, apiv2base.ErrorInvalidFieldFormat, "Cannot rerun more than 50 runs at once")
+	}
+
+	accountID := consts.DevServerAccountID
+	workspaceID := consts.DevServerEnvID
+
+	// Process each run ID
+	results := make([]*apiv2.RerunResult, len(req.RunIds))
```

Comment:
Some of this could be refactored to a cqrs interface or similar to share code between here and the GraphQL API. I'd like to get this out, then plan the interface out for other operations like Replay, cancel, etc.